### PR TITLE
test: add ddl support test for empty model credentials

### PR DESCRIPTION
**Pull request by CodingCookieRookie**
This PR enhances the test coverage for scenarios involving empty credentials in the model handling logic. The v_model view build a lot of information about a model’s credential that is done with a left join. We need a test to assert that when the credential is null in the database the model and it’s credential information correctly come back out of the v_model view..

## Checklist

~~- [ ] Code style: imports ordered, good names, simple structure, etc~~
~~- [ ] Comments saying why design decisions were made~~
- [ ] Go unit tests, with comments saying what you're testing
~~- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~~
~~- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~~

## QA steps

 1. From project directory, go to domain/model/state folder
```sh
cd domain/model/state
```

 2. Run unit tests in the directory
```sh
go test -v .
```

## Links

**Jira card:** https://warthogs.atlassian.net/browse/JUJU-7395

### CODE REVIEW

File: domain/model/state/controllerstate_test.go [hunk 1]
```diff
@@ -1449,6 +1449,56 @@ UPDATE model SET life_id = 1 WHERE uuid = ?
 	})
 }
 
+// TestGetEmptyCredentialsModel verifies that we can retrieve the correct model
+// information even when the model credentials are empty, for both IAAS and Container model types.
```
Comment (tlm) [inline domain/model/state/controllerstate_test.go:1453]: Thanks @CodingCookieRookie, can you expand on this comment a little more for me please?

Be explicit in what we are trying to test. That is we want to make sure the model view continues to work when a credential is nil and potential a join in the sql stops working.

It easy for us to remember why we did this today but in a few years it might not be so clear to the reader why this test exists.

**Review by tlm — APPROVED** [review]
Thanks, approved pending review comment.

File: domain/model/state/controllerstate_test.go (hunk 1 above)
Comment (CodingCookieRookie) [reply domain/model/state/controllerstate_test.go]: Sure thing thanks tom :)

**Review by wallyworld — APPROVED** [review]

**Review by SimonRichardson — COMMENTED** [review]
The QA steps should have added a regression test. 

i.e. `juju add-model`

**Comment by CodingCookieRookie** [comment]
> The QA steps should have added a regression test.
> 
> i.e. `juju add-model`

Hey Simon, thanks for the feedback, would you mind elaborating how would this affect the test suite or vice-versa?

**Comment by CodingCookieRookie** [comment]
> Even though this is just tests, I'd like to ensure that you did bootstrap and create a model, ultimately it gives me confidence that you know the areas of code that you're touching.

I see, sure will write down the qa steps before merging, thanks simon.
